### PR TITLE
HTCONDOR-874 Improve SEC_TOKEN_REVOCATION_EXPR context

### PR DESCRIPTION
**Pull request by jasoncpatton**
Adds idtoken header claims to the list of attributes available in `SEC_TOKEN_REVOCATION_EXPR`

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [x] Check for correctness of change
- [x] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [x] Check for documentation, if needed
- [x] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/condor_io/condor_auth_passwd.cpp [hunk 1]
```diff
@@ -950,6 +950,39 @@ Condor_Auth_Passwd::isTokenRevoked(const jwt::decoded_jwt &jwt)
 		return false;
 	}
 	classad::ClassAd ad;
+	auto claims = jwt.get_header_claims();
+	for (const auto &pair : claims) {
```
Comment (Todd-L-Miller) [inline src/condor_io/condor_auth_passwd.cpp:954]:
```suggestion
	auto header_claims = jwt.get_header_claims();
	for (const auto &pair : header_claims) {
```

Comment (Todd-L-Miller) [reply src/condor_io/condor_auth_passwd.cpp:954]: You can't declare claims twice, and this is clearer anyway.

**Review by Todd-L-Miller — COMMENTED** [review]
Given that the change didn't even build, I'd really like to see a test verifying that we actually put the attributes we say we do in the context ad.

**Comment by Todd-L-Miller** [comment]
I would also like to see any improvement to [the documentation here](https://github.com/htcondor/htcondor/pull/341/files#diff-058ac2d0b926a5d2d335136f2b649e2bc1f758499c32862b6549a80e9ea71c5cL1402-L1403) that makes it explicit that every field in the header and payload becomes an ClassAd attribute with the same name.

**Review by Todd-L-Miller — COMMENTED** [review]
Nice work sliding this into the existing test.

I'll approve once the BaTLab tests pass.

**Comment by GregThain** [comment]
Closed for now as per @jasoncpatton Needs a test to verify.

**Pull request closed without merging**